File: 3163. String Compression III.py
class Solution:
    def compressedString(self, word: str) -> str:
        count, ans, prev = 0, "", word[0]
        for ch in word:
            if ch != prev or count == 9:
                ans += str(count) + prev
                prev = ch
                count = 0
            count += 1

        if count:
            ans += str(count) + prev

        return ans

File: 3163. String Compression III.cpp
class Solution {
public:
    string compressedString(string word) 
    {
        string ans;

        int count = 0;
        char prev = word[0];
        for(auto ch : word)
        {
            if(prev != ch or count == 9)
            {
                ans += to_string(count);
                ans += prev;
                prev = ch;
                count = 0;
            }
            count += 1;
        }
        if(count)
        {
            ans += to_string(count);
            ans += prev;
        }
        return ans;
    }
};